### PR TITLE
Add validation for VK_EXT_present_id and VK_EXT_present_wait

### PR DESCRIPTION
**Pull request by Tony-LunarG**

### CODE REVIEW

File: tests/vklayertests_wsi.cpp [hunk 1]
```diff
@@ -1695,3 +1695,177 @@ TEST_F(VkLayerTest, WarningSwapchainCreateInfoPreTransform) {
...
+        printf("%s surface extensions not supported, skipping test\n", kSkipPrefix);
+        return;
+    }
+    auto present_id_features = LvlInitStruct<VkPhysicalDevicePresentIdFeaturesKHR>();
+    auto present_wait_features = LvlInitStruct<VkPhysicalDevicePresentWaitFeaturesKHR>(&present_id_features);
+    auto features2 = LvlInitStruct<VkPhysicalDeviceFeatures2KHR>(&present_wait_features);
+    m_device_extension_names.push_back(VK_KHR_PRESENT_WAIT_EXTENSION_NAME);
+    m_device_extension_names.push_back(VK_KHR_PRESENT_ID_EXTENSION_NAME);
+    m_device_extension_names.push_back(VK_KHR_SWAPCHAIN_EXTENSION_NAME);
+    bool retval = InitFrameworkAndRetrieveFeatures(features2);
```
Comment (ncesario-lunarg) [inline tests/vklayertests_wsi.cpp:1716]: I keep forgetting this function exists!

File: tests/vklayertests_wsi.cpp [hunk 2]
```diff
@@ -1695,3 +1695,177 @@ TEST_F(VkLayerTest, WarningSwapchainCreateInfoPreTransform) {
     m_errorMonitor->VerifyFound();
     DestroySwapchain();
 }
+
+TEST_F(VkLayerTest, PresentIdWait) {
+    TEST_DESCRIPTION("Test present wait extension");
+    uint32_t version = SetTargetApiVersion(VK_API_VERSION_1_1);
+    if (version < VK_API_VERSION_1_1) {
```
Comment (ncesario-lunarg) [inline tests/vklayertests_wsi.cpp:1702]: Do we need device or just instance version >= 1.1?

**Review by ncesario-lunarg — COMMENTED** [review]
Couple minor nits/questions, but LGTM!

**Review by jeremyg-lunarg — APPROVED** [review]
LGTM

File: layers/core_validation.cpp [hunk 3]
```diff
@@ -13494,6 +13494,33 @@ bool CoreChecks::PreCallValidateQueuePresentKHR(VkQueue queue, const VkPresentIn
                              present_times_info->swapchainCount, pPresentInfo->swapchainCount);
             }
         }
+
+        const auto *present_id_info = LvlFindInChain<VkPresentIdKHR>(pPresentInfo->pNext);
+        if (present_id_info) {
+            if (!enabled_features.present_id_features.presentId) {
+                skip |= LogError(pPresentInfo->pSwapchains[0], "UNASSIGNED-features-VkPresentIdKHR",
```
[the remark this answers is not in the document]
Comment (Tony-LunarG) [reply layers/core_validation.cpp]: Yes, fixed

File: tests/vklayertests_wsi.cpp (hunk 2 above)
Comment (Tony-LunarG) [reply tests/vklayertests_wsi.cpp:1702]: I think instance is enough here

**Review by ncesario-lunarg — APPROVED** [review]

**Comment by Tony-LunarG** [comment]
Build # 4223 passed

**Pull request merged**